changelog/7764.bugfix.md

```diff
@@ -0,0 +1 @@
+Fixes a bug where the next slot asked was not consistent when returning to a form from an unhappy path
```

Comment:
Added the link to our forms documentation so users can jump deeper into the documentation in case they got curious after reading the changelog 
```suggestion
Fixes a bug in [forms](forms.mdx) where the next slot asked was not consistent after returning to a form from an unhappy path.
```